swift-sdk/Internal/ApiClient.swift

```diff
@@ -47,12 +47,12 @@ protocol AuthProvider: class {
 }
 
 class ApiClient: ApiClientProtocol {
-    init(apiKey: String, authProvider: AuthProvider, endPoint: String, networkSession: NetworkSessionProtocol, deviceId: String) {
+    init(apiKey: String, authProvider: AuthProvider, endPoint: String, networkSession: NetworkSessionProtocol, deviceMetadata: DeviceMetadata) {
         self.apiKey = apiKey
         self.authProvider = authProvider
         self.endPoint = endPoint
         self.networkSession = networkSession
-        self.deviceId = deviceId
+        self.deviceMetadata = deviceMetadata
```

Comment:
Hmm I think it's better to move deviceMetadata to request parameters instead of `ApiClient` since device metadata does not apply to all or most of the API calls.